src/ui/server/controllers/insurance/export-contract/currency-of-agents-charge/save-and-back/index.test.ts

```diff
@@ -0,0 +1,121 @@
+import { post } from '.';
+import { INSURANCE_ROUTES } from '../../../../../constants/routes/insurance';
+import INSURANCE_FIELD_IDS from '../../../../../constants/field-ids/insurance';
+import { FIELD_IDS } from '..';
+import constructPayload from '../../../../../helpers/construct-payload';
+import mapAndSave from '../../map-and-save/export-contract-agent-service-charge';
+import generateValidationErrors from '../validation';
+import { Request, Response } from '../../../../../../types';
+import { mockCurrencies, mockReq, mockRes, mockSpyPromiseRejection, referenceNumber } from '../../../../../test-mocks';
+
+const { INSURANCE_ROOT, ALL_SECTIONS, PROBLEM_WITH_SERVICE } = INSURANCE_ROUTES;
+
+const {
+  CURRENCY: { CURRENCY_CODE },
+} = INSURANCE_FIELD_IDS;
+
+describe('controllers/insurance/export-contract/currency-of-agents-charge/save-and-back', () => {
+  let req: Request;
+  let res: Response;
+
+  jest.mock('../../map-and-save/export-contract-agent-service-charge');
+
+  let mapAndSaveSpy = jest.fn(() => Promise.resolve(true));
+
+  const mockFormBody = {
+    _csrf: '1234',
+    [CURRENCY_CODE]: mockCurrencies[0].isoCode,
+  };
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    req.body = mockFormBody;
+
+    mapAndSave.exportContractAgentServiceCharge = mapAndSaveSpy;
+  });
+
+  describe('when the form has data', () => {
```

Comment:
do we have a test for when there are validation errors and when there aren't validation errors like the other save and back tests?

Reply:
Good question, we don't, they are not needed because:

```ts
// save-and-back controller

// ...

if (hasFormData(req.body)) {
  // ...

  mapAndSave.exportContractAgentServiceCharge(); // this handles the validation errors

  return res.redirect();
}

return res.redirect();
```

So that is tested in the `mapAndSave` functions, the "Save and back" controller only cares if there is form data or not, regardless of what state the fields are in